### PR TITLE
NEW - SMBClientCertificateMapping cmdlet

### PR DESCRIPTION
**Pull request by Xelu86**
# PR Summary

This is a new SMB cmdlet, SMBClientCertificateMapping, for WS25.

## PR Checklist

- [x] **Descriptive Title:** This PR's title is a synopsis of the changes it proposes.
- [x] **Summary:** This PR's summary describes the scope and intent of the change.
- [x] **Contributor's Guide:** I have read the [contributors guide][contrib].
- [x] **Style:** This PR adheres to the [style guide][style].

[contrib]: https://learn.microsoft.com/powershell/scripting/community/contributing/overview
[style]: https://learn.microsoft.com/powershell/scripting/community/contributing/powershell-style-guide

### CODE REVIEW

File: docset/winserver2025-ps/smbshare/Get-SmbClientCertificateMapping.md [hunk 1]
```diff
@@ -0,0 +1,291 @@
...
+ [-IssuerName <String[]>] [-Thumbprint <String[]>] [-DisplayName <String[]>] [-StoreName <String[]>]
+ [-Type <Type[]>] [-Flags <Flags[]>] [-IncludeHidden] [-CimSession <CimSession[]>]
+ [-ThrottleLimit <Int32>] [-AsJob] [<CommonParameters>]
+```
+
+## DESCRIPTION
+
+The `Get-SmbClientCertificateMapping` cmdlet retrieves the client certificate mappings for the SMB
+protocol. These mappings are used to authenticate clients that connect to SMB servers using
+certificates.
```
Comment (nedpyle) [inline docset/winserver2025-ps/smbshare/Get-SmbClientCertificateMapping.md:29]: such as when using SMB over QUIC

File: docset/winserver2025-ps/smbshare/Get-SmbClientCertificateMapping.md [hunk 2]
```diff
@@ -0,0 +1,291 @@
...
+
+## EXAMPLES
+
+### Example 1: Retrieve all certificate mappings for a specific store
+
+```powershell
+Get-SmbClientCertificateMapping -StoreName "MyCertificateStore"
```
Comment (nedpyle) [inline docset/winserver2025-ps/smbshare/Get-SmbClientCertificateMapping.md:36]: "my"

File: docset/winserver2025-ps/smbshare/New-SmbClientCertificateMapping.md [hunk 3]
```diff
@@ -0,0 +1,322 @@
...
+SMB client name.
+
+## EXAMPLES
+
+### Example 1: Create a new certificate mapping for an SMB client
+
+```powershell
+$params = @{
+    Thumbprint = "a1b2c3d4e5f6g7h8"
+    StoreName = "MyCertificateStore"
```
Comment (nedpyle) [inline docset/winserver2025-ps/smbshare/New-SmbClientCertificateMapping.md:39]: my

File: docset/winserver2025-ps/smbshare/New-SmbClientCertificateMapping.md [hunk 4]
```diff
@@ -0,0 +1,322 @@
...
+Required: False
+Position: Named
+Default value: None
+Accept pipeline input: False
+Accept wildcard characters: False
+```
+
+### -Namespace
+
+Specifies the namespace in which the certificate mappings are located. By default, the cmdlet
```
Comment (nedpyle) [inline docset/winserver2025-ps/smbshare/New-SmbClientCertificateMapping.md:165]: specifies the namespace of the QUIC server. Example would be "server1.contoso.com"

File: docset/winserver2025-ps/smbshare/Remove-SmbClientCertificateMapping.md [hunk 5]
```diff
@@ -0,0 +1,386 @@
...
+the SMB protocol. These mappings are used to authenticate clients that connect to SMB servers using
+certificates. You can use this cmdlet to remove a mapping between a client certificate and an SMB
+client name.
+
+## EXAMPLES
+
+### Example 1: Remove all certificate mappings from a specific store
+
+```powershell
+Remove-SmbClientCertificateMapping -StoreName "MyCertificateStore" -Force
```
Comment (nedpyle) [inline docset/winserver2025-ps/smbshare/Remove-SmbClientCertificateMapping.md:48]: my

File: docset/winserver2025-ps/smbshare/Remove-SmbClientCertificateMapping.md [hunk 6]
```diff
@@ -0,0 +1,386 @@
...
+Aliases:
+
+Required: False
+Position: Named
+Default value: None
+Accept pipeline input: False
+Accept wildcard characters: False
+```
+
+### -IncludeHidden
```
Comment (nedpyle) [inline docset/winserver2025-ps/smbshare/Remove-SmbClientCertificateMapping.md:155]: not sure if this is true, I will get back to you

File: docset/winserver2025-ps/smbshare/Remove-SmbClientCertificateMapping.md [hunk 7]
```diff
@@ -0,0 +1,386 @@
...
+Required: False
+Position: Named
+Default value: None
+Accept pipeline input: False
+Accept wildcard characters: False
+```
+
+### -StoreName
+
+Specifies the path to the certificate store for the certificate. The recommended value is "My" for
```
Comment (nedpyle) [inline docset/winserver2025-ps/smbshare/Remove-SmbClientCertificateMapping.md:239]: aha! :)

**Review by nedpyle — CHANGES_REQUESTED** [review]
some changes, plus I need to clarify hidden again

File: docset/winserver2025-ps/smbshare/Remove-SmbClientCertificateMapping.md [hunk 8]
```diff
@@ -0,0 +1,386 @@
...
+Required: False
+Position: Named
+Default value: None
+Accept pipeline input: False
+Accept wildcard characters: False
+```
+
+### -DisplayName
+
+Specifies a friendly name to display for the mapping.
```
Comment (nedpyle) [inline docset/winserver2025-ps/smbshare/Remove-SmbClientCertificateMapping.md:102]: The friendly name of the certificate.

File: docset/winserver2025-ps/smbshare/Get-SmbClientCertificateMapping.md [hunk 9]
```diff
@@ -0,0 +1,291 @@
...
+Required: False
+Position: Named
+Default value: None
+Accept pipeline input: False
+Accept wildcard characters: False
+```
+
+### -DisplayName
+
+Specifies a friendly name to display for the mapping.
```
Comment (nedpyle) [inline docset/winserver2025-ps/smbshare/Get-SmbClientCertificateMapping.md:89]: The friendly name of the certificate.

File: docset/winserver2025-ps/smbshare/New-SmbClientCertificateMapping.md [hunk 10]
```diff
@@ -0,0 +1,322 @@
...
+Required: False
+Position: Named
+Default value: None
+Accept pipeline input: False
+Accept wildcard characters: False
+```
+
+### -DisplayName
+
+Specifies a friendly name to display for the mapping.
```
Comment (nedpyle) [inline docset/winserver2025-ps/smbshare/New-SmbClientCertificateMapping.md:94]: The friendly name of the certificate.

**Review by nedpyle — CHANGES_REQUESTED** [review]

File: docset/winserver2025-ps/smbshare/Remove-SmbClientCertificateMapping.md (hunk 6 above)
Comment (robinharwood) [reply docset/winserver2025-ps/smbshare/Remove-SmbClientCertificateMapping.md:155]: @nedpyle - as discussed

Comment (nedpyle) [reply docset/winserver2025-ps/smbshare/Remove-SmbClientCertificateMapping.md:155]: This parameter does nothing and we will remove it. For now can say "This parameter is not used" or to that effect

**Pull request merged**